src/app/src/App.js

```diff
@@ -8,13 +8,42 @@ import Header from './Header';
 import Footer from './Footer';
 import SensorOverview from './SensorOverview';
 
+import { setSensorData } from './app.actions';
+import { POLLING_INTERVAL, SENSORS } from './constants';
+
+import {
+    makeRiverGaugeRequest,
+    parseRiverGaugeApiData,
+    parseRiverGaugeCsvData,
+} from './sensorUtils';
+
 class App extends Component {
+    componentDidMount() {
+        // Poll for new sensor data immediately and on a timed cycle thereafter
+        setInterval(function pollData() {
+            (async () => {
+                const requests = SENSORS.map(sensor =>
+                    makeRiverGaugeRequest(sensor.id, sensor.apiAccess)
```

Comment:
While developing this, did you run into a scenario where one of the sensors was down and failed to retrieve data? I'm curious if one being down would break the fetching of all of the sensors. This may have to be structured so that the sensors can update the state independently of each other as opposed to waiting for all of them to complete.

Reply:
Good questions, I did not run into that case but I'll see what I can do to force an error.

Reply:
Requests are made separately in 660e024 so this should no longer be a concern 👍A good one though!